addons/account/models/account_move.py

```diff
@@ -3251,7 +3251,7 @@ def button_draft(self):
             move.mapped('line_ids.analytic_line_ids').unlink()
 
         self.mapped('line_ids').remove_move_reconcile()
-        self.write({'state': 'draft', 'is_move_sent': False})
+        self.write({'state': 'draft', 'is_move_sent': False, 'invoice_payment_term_id': False})
```

Comment:
Then when resetting to draft an invoice with a payment term, if we change nothing and confirm it, the payment term is gone and was replaced with the due date. There must be a better way to do this.